unittests/gettestkernels.py

```diff
@@ -22,38 +22,62 @@
 SOFTWARE.
 """
 
+import hashlib
 import os
-import time
-import platform
-import tempfile
-import urllib
-import urllib.error
-import urllib.request
+import shutil
 import sys
-import hashlib
+import tempfile
+import time
+from pathlib import Path
+
+import requests
+from requests import RequestException
 
-try:
-    cwd = os.environ['CSPYCE_TEST_KERNELS']
-except KeyError:
-    cwd = "/tmp" if platform.system() == "Darwin" else tempfile.gettempdir()
 
+def _get_kernel_directory():
+    try:
+        return os.environ['CSPYCE_TEST_KERNELS']
+    except KeyError:
+        pass
+
+    directory = Path(tempfile.gettempdir()) / "cspyce-test-kernels"
+    directory.mkdir(parents=True, exist_ok=True)
+    return str(directory)
+
+
+def _get_test_output_directory():
+    directory = Path(tempfile.gettempdir()) / "test_file_output"
+    directory.mkdir(parents=True, exist_ok=True)
+    return str(directory)
+
+
+KERNEL_DIR = _get_kernel_directory()
+TEST_FILE_DIR = _get_test_output_directory()
 
 def get_kernel_name_from_url(url: str) -> str:
     return url.split("/")[-1]
 
 
 def get_path_from_url(url: str) -> str:
-    return os.path.join(cwd, get_kernel_name_from_url(url))
+    return os.path.join(KERNEL_DIR, get_kernel_name_from_url(url))
 
 
-def cleanup_file(path: str) -> None:
+def cleanup_file(_path: str) -> None:
 # =============================================================================
 #     if os.path.exists(path):
 #         os.remove(path)
 # =============================================================================
     pass
 
-class CassiniKernels(object):
+
+def delete_cache_directory():
```

Comment:
This makes me really nervous to have here. I don't think we should ever have a function that deletes an entire directory tree based on the contents of an environment variable. That could lead to all sorts of malicious behavior.

Reply:
This is from Frank's work on the temporary directories. Should this function be deleted or commented out? It does state that it's only to be used in the event of kernel corruption.

Reply:
Just delete it. If you need to delete a directory, just do it from the command line.

Reply:
The reason I added this was that at least on my machine, the directory is off someplace strange.  It's not obvious what to delete.  In my case, one of the download files got corrupted (I was watching the download from my debugger and interrupted it before the download was completed.). So I had a corrupt file, and no way to say if its corrupt, and no way to figure out where it was.

Reply:
Maybe you can just add a separate program to print the directory name?